README.md

```diff
@@ -830,223 +815,155 @@ With multiversion support, you can have two members where each of them having di
-        return 10;
-    }
+namespace hazelcast {
+    namespace client {
+        namespace serialization {
+            template<>
+            struct hz_serializer<Person> : custom_serializer {
+                static constexpr int32_t get_type_id() noexcept {
+                    return 3;
+                }
 
-    virtual void write(serialization::object_data_output &out, const void *object) {
-        const Musician *csObject = static_cast<const Musician *>(object);
-        const std::string &value = csObject->get_name();
-        int length = (int) value.length();
-        std::vector<hazelcast::byte> bytes;
-        for (int i = 0; i < length; ++i) {
-            bytes.push_back((hazelcast::byte) value[i]);
-        }
-        out.writeInt((int) length);
-        out.write(bytes);
-    }
+                static void write(const Person &object, hazelcast::client::serialization::object_data_output &out) {
+                    out.write(object.name);
+                    out.write(object.male);
+                    out.write(object.age);
+                }
 
-    virtual void *read(serialization::object_data_input &in) {
-        int32_t len = in.readInt();
-        std::ostringstream value;
-        for (int i = 0; i < len; ++i) {
-            value << (char) in.readByte();
+                static Person read(hazelcast::client::serialization::object_data_input &in) {
+                    return Person{in.read<std::string>(), in.read<bool>(), in.read<int32_t>()};
+                }
+            };
         }
-
-        return new Musician(value.str());
     }
-};
-```
-
-Note that the serializer `getHazelcastTypeId` method must return a unique `id` as Hazelcast will use it to lookup the `MusicianSerializer` while it deserializes the object.
-
-You should provide the free function `int getHazelcastTypeId(const MusicianSerializer *);` in the same namespace to which `MusicianSerializer` class belongs. This function should return the same id with its serializer. This id is used to determine which serializer needs to be used for your classes. 
-
-Now the last required step is to register the `MusicianSerializer` to the configuration.
-
-```C++
-    client_config clientConfig;
-    clientConfig.get_serialization_config().registerSerializer(
-            boost::shared_ptr<serialization::StreamSerializer>(new MusicianSerializer()));
-
+}
 ```
 
-From now on, Hazelcast will use `MusicianSerializer` to serialize `Musician` objects.
+`hz_serializer<Person>` specialization of `hz_serializer` should implement the above three methods, namely `get_type_id`, `write`, `read`. In case that the object fields are non-public, you can always define struct `hz_serializer<Person>` as friend to your object class. You use the `object_data_output` class methods when serializing the object into binary bytes and you use the `object_data_input` class methods when de-serializing the bytes into the concrete object instance. 
```

Comment:
... as a friend to your ....

... into binary bytes and the `object_data_input` class ...